components/api/src/error.rs

```diff
@@ -14,10 +14,85 @@
 
 //! The set of errors for the Exonum API module.
 
+use actix_web::http::StatusCode;
 use failure::Fail;
 use serde::Serialize;
+
 use std::io;
 
+/// API HTTP error struct.
+#[derive(Fail, Debug, Serialize)]
+pub struct ApiError {
+    /// HTTP error code.
+    #[serde(skip)]
+    pub http_code: StatusCode,
+    /// A URI reference to the documentation or possible solutions for the problem.
+    #[serde(rename = "type", default, skip_serializing_if = "String::is_empty")]
+    pub docs_uri: String,
+    /// Short description of the error.
+    #[serde(default, skip_serializing_if = "String::is_empty")]
```

Comment:
And one more nit: since `ApiError` doesn't derive `Deserialize`, `default` attribute is redundant.